app/src/test/java/com/example/main_animal_villa/CharacterTests.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.main_animal_villa
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.animal_villa.MainViewModel
+import com.animal_villa.dto.Character
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.impl.annotations.MockK
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.*
+import org.junit.rules.TestRule
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+class CharacterTests {
+
+    @get:Rule
+    var rule : TestRule = InstantTaskExecutorRule()
+    var character : Character? = Character()
+
+    lateinit var characterService : CharacterService
+
+    lateinit var mvm : MainViewModel
+
+    @MockK
+    lateinit var mockCharacterService : CharacterService
+
+    private val mainThreadSurrogate = newSingleThreadContext("Main Thread")
+
+    @Before
+    fun initMocksAndMainThread()    {
+        MockKAnnotations.init(this)
+        Dispatchers.setMain(mainThreadSurrogate)
+    }
+
+    @After
+    fun tearDown()  {
+        Dispatchers.resetMain()
+        mainThreadSurrogate.close()
+    }
+
+    @Test
+    fun `given character data is available when I name my character Billy and give them the janitor occupation then Character class should have Billy as name and Janitor as occupation` () = runTest {
+        givenCharacterServiceIsInitialized()
+        whenCharacterDataAreReadAndParsed()
+        thenTheCharacterShouldHaveTheNameBillyAndTheOccupationJanitor()
+    }
+
```

Comment:
An Example given/when/then scenario I have made as an example. Breaking all the different steps down into separate functions allow you to iterate through the different steps more easily, and also make your code reusable for other tests. This is the function you will run to test the entire scenario. The backticks for the function make it a bit more readable, but be sure not to use it in actual code as it is against Kotlin coding conventions. https://kotlinlang.org/docs/coding-conventions.html